desc/objectives/_bootstrap.py

```diff
@@ -0,0 +1,261 @@
+        if verbose > 1:
+            timer.disp("Precomputing transforms")
+
+        if self._normalize:
+            scales = compute_scaling_factors(eq)
+            self._normalization = scales["B"] * scales["J"]
+
+        super().build(eq=eq, use_jit=use_jit, verbose=verbose)
+
+    def compute(self, *args, **kwargs):
+        """Compute the bootstrap current self-consistency objective.
+
+        Returns
+        -------
+        obj : ndarray
+            Bootstrap current self-consistency residual on each rho grid point.
+
+        """
+        params = self._parse_args(*args, **kwargs)
+        data = compute_fun(
+            self._data_keys,
+            params=params,
+            transforms=self._transforms,
+            profiles=self._profiles,
+            helicity=self.helicity,
+        )
+
+        return compress(
+            self.grid, data["<J*B>"] - data["<J*B> Redl"], surface_label="rho"
+        )
+
+    def compute_scaled(self, *args, **kwargs):
+        """Compute and apply the target/bounds, weighting, and normalization."""
+        w = compress(self.grid, self.grid.spacing[:, 0], surface_label="rho")
+        return super().compute_scaled(*args, **kwargs) * jnp.sqrt(w)
+
+    def print_value(self, *args, **kwargs):
+        """Print the value of the objective."""
+        f = self.compute(*args, **kwargs)
+        print("Maximum " + self._print_value_fmt.format(jnp.max(f)) + self._units)
+        print("Minimum " + self._print_value_fmt.format(jnp.min(f)) + self._units)
+        print("Average " + self._print_value_fmt.format(jnp.mean(f)) + self._units)
+
+        if self._normalize:
+            print(
+                "Maximum "
+                + self._print_value_fmt.format(jnp.max(f / self.normalization))
+                + "(normalized)"
+            )
+            print(
+                "Minimum "
+                + self._print_value_fmt.format(jnp.min(f / self.normalization))
+                + "(normalized)"
+            )
+            print(
+                "Average "
+                + self._print_value_fmt.format(jnp.mean(f / self.normalization))
+                + "(normalized)"
+            )
+
+    @property
+    def helicity(self):
+        """tuple: Type of quasi-symmetry (M, N)."""
+        return self._helicity
+
+    @helicity.setter
+    def helicity(self, helicity):
+        assert (
+            (len(helicity) == 2)
+            and (int(helicity[0]) == helicity[0])
+            and (int(helicity[1]) == helicity[1])
+        )
+        assert helicity[0] == 1, "Redl bootstrap current model assumes helicity[0] == 1"
+        self._helicity = helicity
+        if hasattr(self, "_print_value_fmt"):
+            units = "(T)"
+            self._print_value_fmt = (
+                "Quasi-symmetry ({},{}) Boozer error: ".format(
+                    self.helicity[0], self.helicity[1]
+                )
+                + "{:10.3e} "
+                + units
+            )
```

Comment:
For this setter that you added, I wasn't sure if you would rather have this section deleted vs edited to fix units and print bootstrap instead of QS/Boozer.

Reply:
Good catch, I forget to delete that when I copied it over from the QS objectives.